### DISCUSSION TITLE
3.5.10-patch-35303 RavenDB-15974 change reason?

### DISCUSSION THREAD

**Discussion by mthird** [ravendb/ravendb#14280] [Q&A] [answered]
We discovered a major (to us) change in how our Lucene queries were being handled that occurred with 35303 that changed how the following index returned results.   I didn't see any details at https://issues.hibernatingrhinos.com/issue/RavenDB-15974.  Was the old query behavior broken for our case or is this an unintended side-effect?

```cs
public SitesByTags()
        {

            Map = sites => from site in sites
                           select new
                           {
                               Tags = new List<string>(){
                                       $"domain:{site.Domain}",
                                       $"siteid:{site.SiteId}",
                                       $"programDocId:{site.ProgramDocumentId}",
                                       $"dealergroup:{site.DealerGroup}",
                                       $"farm:{site.Farm}",
                                       $"industry:{site.Industry}"
                                   }
                               .Union(site.TeamTags)
                               .Union(site.Channels.Select(x => $"channel:{x.Key}"))
                               .Union(site.Channels.Select(x => $"channelsource:{x.ChannelSource}").Distinct())
                               .Union(site.ExternalAccountIds.Select(x => $"externalaccountid:{x.Value}")),
                               Tenant = site.Tenant
                           };

            this.Index(x => x.Tags, FieldIndexing.Analyzed);
            this.Index(x => x.Tenant, FieldIndexing.NotAnalyzed);
        }
```

With Analyzed, we get a single result on 35302:
![image](https://user-images.githubusercontent.com/537988/170294416-57c660ac-2673-479e-9ddc-9dcfe7a57acd.png)

We get multiple results on 35303+:
![image](https://user-images.githubusercontent.com/537988/170291844-ac134910-da6f-4c16-85c1-8a76fb533bcd.png)

With Default indexing, we get a single result on 35303+:
![image](https://user-images.githubusercontent.com/537988/170292675-287855e1-f1da-4aab-bcf5-7e470f4ebe05.png)

**Comment by ayende** [comment, marked as answer]
The issue is a bit complex, let's see if I can explain. If you have a field that has an analyzer on it, what happens when it emits multiple terms?

Consider: `Name: "Oren Eini"` 

However, this is using an analyzer that breaks it to `["Oren", "Eini"]` terms. 

Previously, this would emit a phrase query, which works, usually. However, things become more complex when your query and the text aren't a match. 

The field may contain stop words, such as `"Oren and Eini"` - which is removed during processing, but then this isn't a match (there is no `Oren` followed by `Eini`), which the individual terms appear. This is why we made this change to a query like that is done using term match.

If you want the old behavior, you can use an explicit phrase query using `Name: "Oren Eini"`
